romancal/resample/meta_blender.py

```diff
@@ -1,8 +1,82 @@
+from collections import defaultdict
+
 import numpy as np
-from roman_datamodels import datamodels, maker_utils
+from asdf.lazy_nodes import AsdfDictNode, AsdfListNode
+from astropy.table import QTable
+from roman_datamodels import datamodels, maker_utils, stnode
+
+
+class MissingCellType:
+    def __str__(self):
+        return "MISSING_CELL"
+
+
+MISSING_CELL = MissingCellType()
+
+
+class TableBuilder:
+    def __init__(self):
+        self._columns = {}
+        self._flagged_columns = set()
+
+    def _get_column(self, name, row_index):
+        if name not in self._columns:
+            if row_index:
+                self._flagged_columns.add(name)
+            self._columns[name] = [MISSING_CELL] * row_index
+        return self._columns[name]
+
+    def _santize_value(self, value):
+        if isinstance(value, list | dict | AsdfDictNode | AsdfListNode):
+            return str(value)
+        return value
+
+    def add_row(self, data_dict, row_index):
+        updated = set()
+        for key, value in data_dict.items():
+            svalue = self._santize_value(value)
+            if svalue is None:
+                self._flagged_columns.add(key)
+            self._get_column(key, row_index).append(svalue)
+            updated.add(key)
+        for key in self._columns.keys() - updated:
+            self._flagged_columns.add(key)
+            self._columns[key].append(MISSING_CELL)
+
+    def to_table(self):
+        array_columns = {}
+        for name, values in self._columns.items():
+            if name not in self._flagged_columns:
+                array_columns[name] = values
+                continue
+            # this column has either a None or MISSING_CELL which
+            # has to be filled in
+            arr = np.array([v for v in values if v not in (None, MISSING_CELL)])
+
+            # if we have no valid values, return all "None"
+            if not arr.size:
+                array_columns[name] = ["None"] * len(values)
+                continue
+
+            # if we have a float or int use 'nan' for missing/None values
+            # this will convert int columns to floats
+            if arr.dtype.kind in ("f", "i"):
+                array_columns[name] = [
+                    np.nan if v in (MISSING_CELL, None) else v for v in values
+                ]
+                continue
+
+            # if all else fails, convert everything to a string
+            array_columns[name] = [str(v) for v in values]
+
+        return QTable(array_columns)
```

Comment:
`QTable` is used here to match what `append_individual_meta` is returning. I couldn't find any columns with units.

Reply:
Thanks.  We probably (?) had a bunch of units in there before the unit removal push a build or two ago, so these could probably go to Tables now.

Reply:
I switched this to `Table` I expect there will be some regtest differences (rerunning them now).

Reply:
I saw no differences. It looks like the [table comparison](https://github.com/spacetelescope/romancal/blob/002371b1f2d5400f8ae79a3fe110e8f1e59e581a/romancal/regtest/regtestdata.py#L608) doesn't distinguish between `Table` and `QTable`.